packages/components/src/DataList/components/DataListItem/DataListItem.tsx

```diff
@@ -0,0 +1,104 @@
+import {
+  DataListLayoutProps,
+  DataListObject,
+} from "@jobber/components/DataList/DataList.types";
+import { InternalDataListItemActions } from "@jobber/components/DataList/components/DataListItemActions";
+import { DataListActionsMenu } from "@jobber/components/DataList/components/DataListActionsMenu";
+import { InternalDataListAction } from "@jobber/components/DataList/components/DataListAction";
+import { DataListLayoutActionsContext } from "@jobber/components/DataList/components/DataListLayoutActions/DataListLayoutContext";
+import styles from "@jobber/components/DataList/DataList.css";
+import { DataListItemInternal } from "./DataListItemInternal";
+import { generateListItemElement } from "../../DataList.utils";
+
+interface DataListItem<T extends DataListObject> {
+  readonly item: T;
+  readonly index: number;
+  readonly layout: ReactElement<DataListLayoutProps<T>>;
+}
+
+export function DataListItem<T extends DataListObject>({
+  item,
+  layout,
+}: DataListItem<T>) {
+  const { itemActionComponent } = useDataListContext();
+  const { hasInLayoutActions } = useDataListLayoutContext();
+  const [showMenu, setShowMenu] = useState(false);
+  const [contextPosition, setContextPosition] =
+    useState<Record<"x" | "y", number>>();
+
+  const generatedItem = useMemo(() => generateListItemElement(item), [item]);
+
+  const contextMenuActions = itemActionComponent?.props.children;
+  const isContextMenuVisible = Boolean(contextPosition);
+  const shouldShowContextMenu =
+    showMenu && isContextMenuVisible && Boolean(contextMenuActions);
+
+  return (
+    <DataListLayoutActionsContext.Provider value={{ activeItem: item }}>
+      <div
+        onMouseEnter={handleShowMenu}
+        onMouseLeave={handleHideMenu}
+        onContextMenu={handleContextMenu}
+        className={classNames(styles.listItem, {
+          [styles.active]: showMenu && isContextMenuVisible,
+        })}
+        key={item.id}
+      >
+        <DataListItemInternal item={item}>
+          {layout.props.children(generatedItem)}
+        </DataListItemInternal>
+
+        <AnimatePresence>
+          {showMenu && !hasInLayoutActions && (
+            <InternalDataListItemActions item={item} />
+          )}
+
+          <DataListActionsMenu
+            key={item.id}
+            visible={shouldShowContextMenu}
+            position={contextPosition || { x: 0, y: 0 }}
+            onRequestClose={() => setContextPosition(undefined)}
+          >
+            {contextMenuActions &&
+              Children.map(contextMenuActions, action => (
+                <InternalDataListAction
+                  key={item.id}
+                  {...action.props}
+                  item={item}
+                />
+              ))}
+          </DataListActionsMenu>
+        </AnimatePresence>
+      </div>
+    </DataListLayoutActionsContext.Provider>
+  );
+
+  function handleShowMenu() {
```

Comment:
I might have missed it - but should we test these event handlers?  
Maybe testing mouse enter/leave and checking that handleShowMenu is called, etc

Reply:
hmmm thought it was tested in the list items with an s, but it is not. I'll save that baby up on the testing task JOB-78870